core/src/main/scala/org/apache/spark/status/api/v1/AllStagesResource.scala

```diff
@@ -59,6 +59,9 @@ private[v1] object AllStagesResource {
       stageUiData: StageUIData,
       includeDetails: Boolean): StageData = {
 
+    val firstTaskLaunchedTime =
+      new Date(stageUiData.taskData.values.map(_.taskInfo.launchTime).filter(_ > 0).min)
```

Comment:
probably worth adding a unit test too, since its easy to forget about this in the future.